src/controllers/admin.js

```diff
@@ -1,125 +1,154 @@
 const mongoose = require('mongoose');
 const Admin = require('../models/Admin');
 
-const createAdmin = (req, res) => {
+const createAdmin = async (req, res) => {
   const {
     firstName, lastName, dni, phone, email, city, password,
   } = req.body;
-  Admin.findOne({ dni })
-    .then((found) => {
-      if (found) {
-        return res.status(404).json({
-          message: `An admin with DNI ${dni} already exists`,
-          data: undefined,
-          error: true,
-        });
-      }
-      return Admin.create({
-        firstName, lastName, dni, phone, email, city, password,
+  try {
+    const found = await Admin.findOne({ dni });
+    if (found) {
+      return res.status(404).json({
+        message: `An admin with DNI ${dni} already exists`,
+        data: undefined,
+        error: true,
       });
-    })
-    .then((result) => res.status(201).json({
+    }
+    const result = await Admin.create({
+      firstName, lastName, dni, phone, email, city, password,
+    });
+    return res.status(201).json({
       message: 'New admin created',
       data: result,
       error: false,
-    }))
-    .catch((error) => res.status(400).json({
-      message: 'An error ocurred!',
-      error,
-    }));
+    });
+  } catch (error) {
+    return res.status(400).json({
+      message: error.message,
+      data: undefined,
+      error: true,
+    });
+  }
 };
 
 const updateAdmin = async (req, res) => {
   const { id } = req.params;
   const {
     firstName, lastName, dni, phone, email, city, password,
   } = req.body;
-  if (!mongoose.Types.ObjectId.isValid(id)) {
-    return res.status(404).json({
-      message: 'Id is not a valid one',
-      data: undefined,
-      error: true,
+
+  try {
+    if (!mongoose.Types.ObjectId.isValid(id)) {
+      return res.status(404).json({
+        message: 'Id is not a valid one',
+        data: undefined,
+        error: true,
+      });
+    }
+
+    const actualAdmin = await Admin.findById(id);
+    const adminProperties = Object.keys(actualAdmin.toObject()).slice(1, -1);
```

Comment:
You should check here if the admin with that id is found before doing this logic, instead of doing it on line 80, like:
```
if (!actualAdmin) {
      return res.status(404).json({
        message: `Admin with id ${id} was not found`,
        data: undefined,
        error: true,
      });
    }
```

Reply:
Yesss you have a good eye